src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-snapshotschedule-form/cephfs-snapshotschedule-form.component.ts

```diff
@@ -91,25 +101,52 @@ export class CephfsSnapshotscheduleFormComponent extends CdForm implements OnIni
     this.directoryStore.loadDirectories(this.id, '/', 3);
     this.createForm();
     this.isEdit ? this.populateForm() : this.loadingReady();
-    this.snapScheduleForm.get('directory').valueChanges.subscribe({
-      next: (value: string) => {
-        this.subvolumeGroup = value?.split?.('/')?.[2];
-        this.subvolume = value?.split?.('/')?.[3];
-        this.subvolumeService
-          .exists(
-            this.subvolume,
-            this.fsName,
-            this.subvolumeGroup === DEFAULT_SUBVOLUME_GROUP ? '' : this.subvolumeGroup
-          )
-          .subscribe({
-            next: (exists: boolean) => {
-              this.isSubvolume = exists;
-              this.isDefaultSubvolumeGroup =
-                exists && this.subvolumeGroup === DEFAULT_SUBVOLUME_GROUP;
+    !this.isEdit &&
+      this.snapScheduleForm
+        .get('directory')
+        .valueChanges.pipe(debounceTime(DEBOUNCE_TIMER))
+        .subscribe({
+          next: (value: string) => {
+            this.subvolumeGroup = value?.split?.('/')?.[2];
+            this.subvolume = value?.split?.('/')?.[3];
+            if (this.subvolume && this.subvolumeGroup) {
+              this.subvolumeService
+                .exists(
+                  this.subvolume,
+                  this.fsName,
+                  this.subvolumeGroup === DEFAULT_SUBVOLUME_GROUP ? '' : this.subvolumeGroup
+                )
+                .pipe(
+                  tap((exists: boolean) => (this.isSubvolume = exists)),
+                  tap(
+                    (exists: boolean) =>
+                      (this.isDefaultSubvolumeGroup =
+                        exists && this.subvolumeGroup === DEFAULT_SUBVOLUME_GROUP)
+                  ),
+                  filter((exists: boolean) => exists),
+                  switchMap(() =>
+                    this.subvolumeService
+                      .info(
+                        this.fsName,
+                        this.subvolume,
+                        this.subvolumeGroup === DEFAULT_SUBVOLUME_GROUP ? '' : this.subvolumeGroup
+                      )
+                      .pipe(pluck('path'))
+                  )
+                )
+                .subscribe({
```

Comment:
is the nested subscription necessary?

Reply:
@nizamial09 
Nested subscription should be avoided. This is now fixed. Thanks